src/terra-dev-site/contributing/Conventions.d.contributing.md

```diff
@@ -0,0 +1,254 @@
+ variant: PropTypes.oneOf(['link', 'outline']),
+};
+
+const defaultProps = {
+ isDisabled: false,
+};
+
+// Component definition
+const Button = (props) => {
+  ...
+}
+
+Button.propTypes = propTypes;
+Button.defaultProps = defaultProps;
+```
+
+#### Keep propTypes as minimal as possible
+
+```jsx
+// good
+propName
+
+// Bad
+longPropNameThatEveryoneMustTypeForever
+
+// Bad
+p
+```
+
+#### Mirror DOM wherever possible
+
+```jsx
+// Good
+className
+
+// Bad
+themeType
+```
+
+```jsx
+// Good
+onClick
+
+// Bad
+whenClicked
+```
+
+#### For boolean props, define defaultProp value as false
+
+Use naming conventions like isOpen, hasValue, isSelected, etc to indicate a true boolean value
+
+```jsx
+// Good
+isLoading
+
+// Bad
+loading
+```
+#### Exceptions
+
+1. If a boolean prop is used to represent an html boolean attribute, use the html attribute name
+
+```jsx
+// Good
+<input required />
+
+// Bad
+<input required={isRequired} />
+```
+
+2. If a subcomponent is designed be used with Higher Order Components with different default behaviors, the defaultProps value could be defaulted to undefined.
+  - For example, a list item can be selectable. A default list wants each list item's isSelectable prop default value to be false, while a selectable list wants each list item's isSelectable prop default value to be true. In this instance, the list item should not provide a default.
+
+### For array props, use plurals
+
+```jsx
+// Good
+items
+
+// Bad
+list
+```
+
+### Denote required props
+
+```jsx
+// Good
+MyComponent.propTypes = {
+  requiredString: React.PropTypes.string.isRequired
+}
+
+// Bad
+MyComponent.propTypes = {
+  requiredString: React.PropTypes.string
+}
+```
+
+## Testing Conventions
+
+Terra UI has 2 types of tests:
+* Unit tests, using [Jest](https://github.com/facebook/jest) and [Enzyme](https://github.com/airbnb/enzyme)
+* Browser tests, using [Webdriver.io](http://webdriver.io/)
+
+
+### Jest And Enzyme
+
+We use Jest for unit tests and snapshot testing for components.
+
+At minimum, all components should have a "does it render" test:
+
+```jsx
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Link from './Link';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  ReactDOM.render(<Link />, div);
+});
+```
+
+If the component accepts classNames, test to ensure it properly merges the passed in the class name with the component class name.
+
+```jsx
+it('should merge classes passed in with attributes', () => {
+  const button = shallow(<Button className="TestClass" text="text" />);
+  expect(button.prop('className')).toContain('TestClass');
+});
+```
+
+You may also want to test class level functions:
+
+```jsx
+it('tracks the number of clicks', () => {
+  const div = document.createElement('div');
+  const instance = ReactDOM.render(<Link />, div);
+
+  t.equals(instace.state.clicks, 0);
+  instance.trackClick();
+  t.equals(instace.state.clicks, 1);
+});
+```
```

Comment:
Should we add documentation for testing i18n components with jest/enzyme? https://gist.github.com/emilyrohrbough/05eb78bb15ee54d416f4761efc589cb3

Reply:
Updated: https://github.com/cerner/terra-ui/pull/108/commits/0120d495f4356a69257518f07acafcb9be89213d